.github/workflows/nightly-release.yaml

```diff
@@ -0,0 +1,99 @@
+# Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+# Exceptions. See /LICENSE for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# This workflow creates a GitHub "release" of a nightly build of the project.
```

Comment:
Should this be accompanied by something like https://github.com/marketplace/actions/delete-releases to limit how many nightly releases remain?